packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -890,6 +890,10 @@ class FlutterError extends Error with DiagnosticableTreeMixin implements Asserti
 
   @override
   String toString({DiagnosticLevel minLevel = DiagnosticLevel.info}) {
+    if (kReleaseMode) {
+      final Iterable<_ErrorDiagnostic> errors = diagnostics.whereType<_ErrorDiagnostic>();
```

Comment:
How much does this increase binary size?  That was the primary driver of this change

Reply:
Ideally we want to completely avoid retaining the diagnostics related classes

Reply:
This will slightly reduce the binary size.  For the sample app in https://github.com/flutter/flutter/issues/54038 the total size of the segments in libapp.so is:
  * 3611038 with this patch
  * 3613222 without

The diagnostics classes are present in the release build, but their `toString` methods return an empty string.  `FlutterError.toString` was passing the diagnostics nodes to the `TextTreeRenderer`, which just yields a series of empty strings.  With this patch `FlutterError.toString` will return the value of the `ErrorSummary`, and the compiler can eliminate the `TextTreeRenderer` invocation.

Reply:
Ahh ok nice